### PR TITLE
World composition singleton improvements

### PR DESCRIPTION
**Pull request by EstevanBR**

### CODE REVIEW

File: CollidableManager.cpp [hunk 1]
```diff
@@ -0,0 +1,29 @@
+//
```
Comment (EstevanBR) [inline CollidableManager.cpp:1]: The World class used to have a vector of collidables, and one of entities. but then I couldn't derive the world class into something that had more customized behaviour in the update function, or various members. So instead I broke it up into CollidableManager and EntityManager.

File: Component.cpp [hunk 2]
```diff
@@ -5,7 +5,7 @@
 #include "Component.hpp"
```
Comment (EstevanBR) [inline Component.cpp:5]: I liked the naming of entity_ more than parent_.

File: ComponentMessaging.hpp [hunk 3]
```diff
@@ -13,21 +13,44 @@
```
Comment (EstevanBR) [inline ComponentMessaging.hpp:13]: adding more message types.

File: ComponentMessaging.hpp [hunk 4]
```diff
@@ -13,21 +13,44 @@
...
     VECTOR2i,
-    STRING
+    STRING,
+    INT,
+    FLOAT,
+    BOOL
 };
 
 template <MESSAGE_TYPES> class Message;
+class Descriptable {
```
Comment (gillett-hernandez) [inline ComponentMessaging.hpp]: describable?

File: PhysicsComponent.hpp [hunk 5]
```diff
@@ -8,26 +8,30 @@
...
-    PhysicsComponent(Entity &entity): Component(entity), world_(*World::instance()), entity_(entity) {
+    PhysicsComponent(Entity &entity): Component(entity) {
 
     }
-    const sf::IntRect *placeFree(int x, int y, sf::Rect<int> rect) {
+    const sf::IntRect *placeFree(const int &x, const int &y, sf::Rect<int> rect) {
         rect.left = x;
         rect.top = y;
-        for (sf::IntRect *collidable : world_.collidables) {
+        assert(world_ != nullptr && "world_ was null. Did you provide it in the Derived Class's Constructor?");
```
Comment (EstevanBR) [inline PhysicsComponent.hpp]: @gillett-hernandez what do you think of this?

File: ComponentMessaging.hpp (hunk 4 above)
Comment (EstevanBR) [reply ComponentMessaging.hpp]: dur

**Review by gillett-hernandez — APPROVED** [review]
looks good. nothing seems weird except for "descriptable" != "describable"

File: Component.cpp (hunk 2 above)
Comment (gillett-hernandez) [reply Component.cpp:5]: `owner` could work too

File: World.cpp [hunk 6]
```diff
@@ -2,57 +2,6 @@
...
-    } else {
-        if (debug == true) {
-            std::cout << "that entity is already in the Worlds entities" << std::endl;
-        }
-    }
-}
-
-void World::removeCollidable(sf::IntRect *rect) {
-    this->collidables.erase(std::remove(this->collidables.begin(), this->collidables.end(), rect), this->collidables.end());
-}
```
Comment (EstevanBR) [inline World.cpp:55]: Feelsgoodman

**Pull request merged**